engine/Default/galactic_post_delete.php

```diff
@@ -5,19 +5,13 @@
 }
 else {
 	//we are deleting an entire paper
-	//make sure it isnt online
-	$db->query('SELECT * FROM galactic_post_online WHERE game_id = ' . $db->escapeNumber($player->getGameID()) . ' AND paper_id = '.$db->escapeNumber($var['id']));
-	if ($db->getNumRows())
-		create_error('You cant delete a paper that is online.  You must first replace the paper with another.');
-	else {
-		//its ok to delete this paper
-		//find out which articles need to be deleted as well
-		$db->query('SELECT * FROM galactic_post_paper_content WHERE game_id = ' . $db->escapeNumber($player->getGameID()) . ' AND paper_id = '.$db->escapeNumber($var['id']));
-		while($db->nextRecord()) {
-			//delete this article that is part of this paper
-			$db2->query('DELETE FROM galactic_post_article WHERE article_id = ' . $db->escapeNumber($db->getInt('article_id')) . ' AND game_id = ' . $db->escapeNumber($player->getGameID()));
-		}
+	//find out which articles need to be deleted as well
+	$db->query('SELECT * FROM galactic_post_paper_content WHERE game_id = ' . $db->escapeNumber($player->getGameID()) . ' AND paper_id = '.$db->escapeNumber($var['id']));
+	while($db->nextRecord()) {
+		//delete this article that is part of this paper
+		$db2->query('DELETE FROM galactic_post_article WHERE article_id = ' . $db->escapeNumber($db->getInt('article_id')) . ' AND game_id = ' . $db->escapeNumber($player->getGameID()));
```

Comment:
You can make this into a single query with:
```sql
DELETE FROM galactic_post_article gpa
WHERE EXISTS (
	SELECT 1
	FROM galactic_post_paper_content gpc
	WHERE gpa.game_id = gpc.game_id
	AND gpa.article_id = gpc.article_id
	AND gpc.game_id = :game_id
	AND gpc.paper_id = :paper_id
)

Reply:
I actually only changed whitespace on these lines, so I'd prefer not to change this query if possible. But I will mark this down as a future change to make, if that's okay.